patches/net/minecraft/world/effect/MobEffect.java.patch

```diff
@@ -9,18 +9,44 @@
      public static final Codec<Holder<MobEffect>> CODEC = BuiltInRegistries.MOB_EFFECT.holderByNameCodec();
      public static final StreamCodec<RegistryFriendlyByteBuf, Holder<MobEffect>> STREAM_CODEC = ByteBufCodecs.holderRegistry(Registries.MOB_EFFECT);
      private static final int AMBIENT_ALPHA = Mth.floor(38.25F);
-@@ -179,6 +_,14 @@
-     @Override
-     public FeatureFlagSet requiredFeatures() {
-         return this.requiredFeatures;
+@@ -130,6 +_,14 @@
+         return this;
+     }
+ 
++    /**
++     * Neo: attribute template with custom level curve
++     */
++    public MobEffect addAttributeModifier(Holder<Attribute> p_316656_, ResourceLocation p_350368_, it.unimi.dsi.fastutil.ints.Int2DoubleFunction p_19475_, AttributeModifier.Operation p_19476_) {
++        this.attributeModifiers.put(p_316656_, new MobEffect.AttributeTemplate(p_350368_, p_19475_.apply(0), p_19476_, p_19475_));
++        return this;
```

Comment:
Since this method is patched-in, it needs a full javadoc, and should not use obfuscated parameter names.

Also, the operation and function parameter order should be switched